lib/events/sessionlog.go

```diff
@@ -213,6 +234,11 @@ func (sl *DiskSessionLogger) WriteChunk(chunk *SessionChunk) (written int, err e
 	sl.Lock()
 	defer sl.Unlock()
 
+	// when session recording is turned off, don't record the session byte stream
+	if sl.recordSessions == false {
+		return 0, nil
```

Comment:
should not return 0, should return lengh written, so implementations of io.Copy using it won't fail